### PR TITLE
MOB-1919 Added the possibility of distributing the app externally

### PR DESCRIPTION
**Pull request by d4r1091**
[MOB-1919](https://ecosia.atlassian.net/browse/MOB-1919)

## Context

At the moment we have an “Ecosia Team” Testflight External Testing team which already contains some Ecosians.
To invite that team, we have to remember to manually invite them to each new version so that a review can be started (since it is external it needs review) and then the testers can be notified after approved.

## Approach

Added the external distribution approach via TestFlight `pilot`

## Other 

⚠️  Want to make a fake build to test it out e2e after we release the 8.3.0

### CODE REVIEW

**Review by lucaschifino — DISMISSED** [review]
Nice, looks good 👏

**Review by ecotopian — DISMISSED** [review]
Neat addendum 💡

**Comment by d4r1091** [comment]
So here is the outcome that also served as reminder to myself.
To perform actions such as inviting users or send apps to testing, we need some setup at CI level. This setup consists into adding the `FASTLANE_PASSWORD` environment variable containing our Ecosia Developer password.
As we do have 2FA (2 step verification) enabled on that account, we also need to generate (via CLI) and provide another environment variable `FASTLANE_SESSION` that lasts up to 30days if not mistaken.

Not sure about the implications (according to Ecosia) security wise of adding the account's psw as CI env var. Also, counting on our release speed cycle (at least lately) we need to take into consideration that we'll be adding another step into the release cycle (generating and adding this  `FASTLANE_SESSION`).

My point of view:
- Counting on all the above and considering our use case, I see the task of inviting testers not a big deal.
- Improvement: Internal Testing is enabled by default, so we could think of moving all the `Ecosia Team` testers as Internal and solve it 🤔

**Comment by d4r1091** [comment]
After awaiting the app for processing, I'm confirming the theory by which we will need to add both `FASTLANE_PASSWORD` and provide with a `FASTLANE_SESSION` on top of incrementing the timeout processing time.
The `FASTLANE_SESSION` will last roughly 30days since last used

**Comment by lucaschifino** [comment]
@d4r1091 good findings, thanks for sharing!

I know that fastlane has support for using the App Store Connect API keys in some actions, which I believe would solve this password and session problem. From [their documentation](https://docs.fastlane.tools/app-store-connect-api/#supported-actionstools) it seems `pilot` supports it, but I have never used it, so not sure if it fits in our case. Did you look into that as well?

If it becomes too much of a hassle to implement anyway, I agree with postponing for now and doing the invitation process manually. I like the suggestion of making the Ecosia Team internal testers as well, since we have only a few people at the moment, but believe this is not very scalable since everyone needs to be added to App Store Connect for that.

**Comment by d4r1091** [comment]
Thanks for sharing it @ecotopian . I guess I could give it a try today with apple-dev account and verify if it works. Will timebox it to 2h max

File: fastlane/Fastfile [hunk 1]
```diff
@@ -110,10 +112,19 @@ platform :ios do
       configuration: "Release"
     )
 
+    app_store_connect_api_key(
+      key_id: ENV["FASTLANE_APPSTORE_CONNECT_KEY_ID"],
```
Comment (ecotopian) [inline fastlane/Fastfile:116]: Could you please provide the values of these env variables in a shared Bitwarden note? That would be great 🙏

**Review by ecotopian — APPROVED** [review]
Excellent work 🚀🚀🚀 Now we just need to advertise our Testflight channel and win more Ecosians to participate 👪

**Review by lucaschifino — APPROVED** [review]
Great job, LGTM 🟢 

Glad the API key worked 🙌

File: fastlane/Fastfile (hunk 1 above)
Comment (d4r1091) [reply fastlane/Fastfile:116]: I wish I could @ecotopian . Actually tried it. Meant to reach out to you to provide you with the .p8 file

**Pull request merged**